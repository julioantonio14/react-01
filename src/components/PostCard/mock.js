export const postCardPropsMock =  {
    title: "alguma coisa",
    cover: "alguma Imgagem",
    body: "Ol√° Mundo",
};
