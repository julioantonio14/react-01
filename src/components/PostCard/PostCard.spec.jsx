import { render,screen } from "@testing-library/react"
import { PostCard } from "."
import { postCardPropsMock } from "./mock"

const mock = postCardPropsMock;
describe('<PostCard />', ()=>{
    it('Should Render PostCard Correctly', ()=>{
        render(<PostCard post={ mock}/>);
        expect(screen.getByRole('img',{name: mock.title})).toBeInTheDocument();
        expect(screen.getByRole('heading',{name: mock.title})).toBeInTheDocument();
    })
    it('Should match snapshot', ()=>{
        const {container} = render(<PostCard post={ mock}/>);
        expect(container.firstChild).toMatchSnapshot();
    })
})