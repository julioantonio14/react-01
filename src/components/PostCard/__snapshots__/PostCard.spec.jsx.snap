// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="alguma coisa"
    src="alguma Imgagem"
  />
  <div
    class="post-content"
  >
    <h2>
      alguma coisa
    </h2>
    <p>
      Ol√° Mundo
    </p>
    <p>
      Oi
    </p>
  </div>
</div>
`;
